node/external/transactionAPI/apiTransactionProcessor.go

```diff
@@ -487,14 +487,35 @@ func (atp *apiTransactionProcessor) recreateTrie(blockchain data.ChainHandler, a
 	rootHashHolder := holders.NewRootHashHolder(currentRootHash, core.OptionalUint32{Value: epoch, HasValue: true})
 
 	// TODO: keep in mind that the selection simulation can be affected by other API requests which might alter the trie
-	err := accountStateAPI.RecreateTrie(rootHashHolder)
+	err = accountStateAPI.RecreateTrie(rootHashHolder)
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
 
+func (atp *apiTransactionProcessor) getCurrentRootHash(
+	blockchain data.ChainHandler,
+) ([]byte, error) {
+	blockHeader := blockchain.GetCurrentBlockHeader()
+	if blockHeader == nil {
+		return nil, ErrNilBlockHeader
+	}
+
+	if blockHeader.IsHeaderV3() {
```

Comment:
when committing the first v3 header, do we create a lastExecutedBlockInfo from the previous V2 block?
otherwise we might have a gap here.